### PR TITLE
ocamlPackages.mirage-block: init at 2.0.1

### PR DESCRIPTION
**Pull request by vbgl**
###### Motivation for this change

#23955

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/mirage-block/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  };
+
+  propagatedBuildInputs = [ cstruct lwt mirage-device ];
+
+  meta = {
+    description = "Block signatures and implementations for MirageOS";
+    homepage = "https://github.com/mirage/mirage-block";
+    license = lib.licenses.isc;
+    maintainers = [ lib.maintainers.vbgl ];
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/mirage-block/default.nix:23]:
```suggestion
  meta = with lib; {
    description = "Block signatures and implementations for MirageOS";
    homepage = "https://github.com/mirage/mirage-block";
    license = licenses.isc;
    maintainers = with maintainers; [ vbgl ];
  };
```

File: pkgs/development/ocaml-modules/mirage-block/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  version = "2.0.1";
+
+  useDune2 = true;
+
+  src = fetchurl {
+    url = "https://github.com/mirage/mirage-block/releases/download/v${version}/mirage-block-v${version}.tbz";
+    sha256 = "1wp8wmixaz9i2sbvq6nkx903lbnpdgb2w404pz1wk8kcg9p3ilcc";
+  };
+
+  propagatedBuildInputs = [ cstruct lwt mirage-device ];
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/mirage-block/default.nix:16]: You are sure this are not only build dependencies?

**Review by SuperSandro2000 — COMMENTED** [review]
Build on darwin:

Result of `nixpkgs-review pr 101825` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ocamlPackages.mirage-block</li>
  </ul>
</details>

File: pkgs/development/ocaml-modules/mirage-block/default.nix (hunk 2 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/mirage-block/default.nix:16]: Yes I am.

File: pkgs/development/ocaml-modules/mirage-block/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/mirage-block/default.nix:23]: What’s the motivation for this change?

Comment (SuperSandro2000) [reply pkgs/development/ocaml-modules/mirage-block/default.nix:23]: Having less lib and making it easier to add other maintainers without code duplication.

**Comment by cole-h** [comment]
@ofborg eval

**Pull request merged**